shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx

```diff
@@ -0,0 +1,175 @@
+
+    ```xml
+    <CheckBox
+        android:id="@+id/MuteLocalAudio"
+        android:text="Mute local audio"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_below="@id/JoinButton"
+        android:layout_alignStart="@id/JoinButton" />
+
+    <CheckBox
+        android:id="@+id/MuteLocalVideo"
+        android:text="Mute local video"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_below="@id/MuteLocalAudio"
+        android:layout_alignStart="@id/JoinButton" />
+    ```
+
+### Handle the system logic
+
+1. **Import supporting libraries**
+
+    In `MainActivity.java`, add the following to the list of `import` statements:
+
+    ```java
+    import java.lang.Math;
+    import android.widget.CheckBox;
+    import android.widget.CompoundButton;
+    ```
+
+1. **Define variables to access checkboxes**
+
+    In `/app/java/com.example.<projectname>/MainActivity`, add the following declarations to the `MainActivity` class:
+
+    ```java
+    private CheckBox checkMuteLocalAudio;
+    private CheckBox checkMuteLocalVideo;
+    ```
+
+### Implement channel quality features
+
+1. **Configure the <Vpd k="SDK" /> log file**
+
+    To customize the location, and content of the log file, add the following code to `setupAgoraRtcService()` before `agoraRtcService.init(appId, agoraRtcEvents, options);`
+
+    ```java
+    // Configure log file location and logging level
+    options.logCfg.logPath = getExternalFilesDir(null).getPath() + "/rtsalog";
+    options.logCfg.logLevel = AgoraRtcService.LogLevel.RTC_LOG_WARNING;
+    ```
+
+1. **Specify the audio codec, sampling rate and the number of channels**
+
+    You set the audio codec, sampling rate, and the number of channels in `ChannelOptions` that you pass to the `agoraRtcService.joinChannel` method. To set these parameters, modify the following lines in the `joinChannel(View view)` method according to your requirements:
+
+    ```java
+    channelOptions.audioCodecOpt.audioCodecType
+            = AgoraRtcService.AudioCodecType.AUDIO_CODEC_TYPE_OPUS;
+    channelOptions.audioCodecOpt.pcmSampleRate = 16000;
+    channelOptions.audioCodecOpt.pcmChannelNum = 1;
+    ```
+  
+    Choose from `OPUS`, `G722`, `G711A`, and `G711U` built-in audio codecs in the <Vpd k="SDK" />. 
+    
+    To use your own audio codec, set `channelOptions.audioCodecOpt.audioCodecType` to `AUDIO_CODEC_DISABLED`. When you call `agoraRtcService.sendAudioData`, set the `dataType` parameter of `AudioFrameInfo` to your encoding format. This setting just transmits the audio encoding format to the receiving end. When you disable use of a built-in audio codec, <Vpd k="SDK" /> does not process the audio. The receiving end obtains the encoded audio data and the encoding format in the `onAudioData` callback and decodes it using a custom decoder.
```

Comment:
This setting transmits the ...


Avoid words like just or simply

Reply:
removed